### DISCUSSION TITLE
calling server actions from server components

### DISCUSSION THREAD

**Discussion by asnaeb** [vercel/next.js#54279] [App Router]
When calling server actions from client, an endpoint is created and a POST request to it is sent with the function arguments as payload. Does the same happens when calling server actions from server components? Apparently it looks like it doesn't but could it be considered good practice? The use case here is having a single `"use server";` file with actions defined in it which could be called from both server components and client components without the need for redeclaring those functions.

**Comment by picardplaisimond** [comment]
I have the same question right now. It's hard to find the answer,

**Comment by Riczer** [comment]
I have the same question.
